akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -80,13 +80,13 @@ abstract class PathMatcher[L](implicit val ev: Tuple[L]) extends (Path ⇒ PathM
     new PathMatcher[lift.Out]()(lift.OutIsTuple) {
       require(min >= 0, "`min` must be >= 0")
       require(max >= min, "`max` must be >= `min`")
-      def apply(path: Path) = rec(path, 1)
+      def apply(path: Path) = rec(path, 0)
       def rec(path: Path, count: Int): Matching[lift.Out] = {
         def done = if (count >= min) Matched(path, lift()) else Unmatched
-        if (count <= max) {
+        if (count < max) {
```

Comment:
check if count is less than max (instead of less than or equal to), because if **count is already max value**, no need to pathmatcher for expected pattern and run below logic.